addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -0,0 +1,388 @@
+                (0, 0, {
+                    'name': aml.name,
+                    'debit': reported_credit,
+                    'credit': reported_debit,
+                    'amount_currency': -reported_amount_currency,
+                    'currency_id': aml.currency_id.id,
+                    'account_id': aml.account_id.id,
+                    'partner_id': aml.partner_id.id,
+                }),
+                (0, 0, {
+                    'name': _('Adjusting Entry'),
+                    'debit': reported_debit,
+                    'credit': reported_credit,
+                    'amount_currency': reported_amount_currency,
+                    'currency_id': aml.currency_id.id,
+                    'account_id': accrual_account.id,
+                    'partner_id': aml.partner_id.id,
+                }),
+            ]
+
+        move_vals = [m for m in move_data.values()]
+        return move_vals
+
+    @api.depends('move_line_ids', 'journal_id', 'journal_id', 'revenue_accrual_account', 'expense_accrual_account', 'percentage', 'date', 'account_type', 'action', 'destination_account_id')
+    def _compute_move_data(self):
+        for record in self:
+            if record.action == 'change_period':
+                if any(line.account_id.user_type_id != record.move_line_ids[0].account_id.user_type_id for line in record.move_line_ids):
+                    raise UserError(_('All accounts on the lines must be of the same type.'))
+            record.move_data = json.dumps(getattr(record, '_get_move_dict_vals_%s' % record.action)())
+
+    @api.depends('move_data')
+    def _compute_preview_move_data(self):
+        for record in self:
+            preview_columns = [
+                {'field': 'account_id', 'label': _('Account')},
+                {'field': 'name', 'label': _('Label')},
+                {'field': 'debit', 'label': _('Debit'), 'class': 'text-right text-nowrap'},
+                {'field': 'credit', 'label': _('Credit'), 'class': 'text-right text-nowrap'},
+            ]
+            if record.action == 'change_account':
+                preview_columns[2:2] = [{'field': 'partner_id', 'label': _('Partner')}]
+
+            move_vals = json.loads(record.move_data)
+            preview_vals = []
+            for move in move_vals[:4]:
+                preview_vals += [self.env['account.move']._move_dict_to_preview_vals(move, record.company_id.currency_id)]
+            preview_discarded = max(0, len(move_vals) - len(preview_vals))
+
+            record.preview_move_data = json.dumps({
+                'groups_vals': preview_vals,
+                'options': {
+                    'discarded_number': _("%d moves", preview_discarded) if preview_discarded else False,
+                    'columns': preview_columns,
+                },
+            })
+
+    def do_action(self):
+        move_vals = json.loads(self.move_data)
+        return getattr(self, '_do_action_%s' % self.action)(move_vals)
```

Comment:
Since you have a binary choice there, call the methods explicitely instead of using a `getattr`